cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -366,7 +367,10 @@ func GetAvailableResources(clientBuilder controller.ControllerClientBuilder) (ma
 
 	resourceMap, err := discoveryClient.ServerResources()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get supported resources from server: %v", err)
+		utilruntime.HandleError(fmt.Errorf("unable to get all supported resources from server: %v", err))
```

Comment:
The TPR issue @mikedanese mentioned is https://github.com/kubernetes/kubernetes/issues/22768#issuecomment-220522337. If user created multiple versions of a TPR, only the first version is operational, so the discovery here fails and causes controller manager to crashloop.